src/main/java/object/piece/Piece.java

```diff
@@ -0,0 +1,74 @@
+package object.piece;
+
+import java.util.List;
+import java.util.Objects;
+import object.coordinate.Position;
+import object.moverule.MoveRule;
+
+public class Piece {
+
+    private final Team team;
+    private final MoveRule moveRule;
+    private final Position position;
+
+    public Piece(Team team, MoveRule moveRule, Position position) {
+        this.team = team;
+        this.moveRule = moveRule;
+        this.position = position;
+    }
+
+    public Piece move(Position from, Position to, List<Piece> piecesOnBoard) {
+        moveRule.checkAbleToMove(from, to, piecesOnBoard, team);
+        return new Piece(this.team, this.moveRule, to);
+    }
+
+    public boolean isSameTeam(Team moveTeam) {
+        return team.equals(moveTeam);
+    }
+
+    public boolean isSameTeam(Piece comparePiece) {
+        return isSameTeam(comparePiece.team);
+    }
+
+    public boolean isSamePosition(Position position) {
+        return this.position.equals(position);
+    }
+
+    public boolean isSamePosition(Piece comparePiece) {
+        return isSamePosition(comparePiece.position);
+    }
```

Comment:
Team, Position 외에 Piece를 넣어줘서 클라이언트 코드에게 편의성을 제공한 건 좋은 시도인 것 같습니다. 하지만 Piece를 인자로 받는 케이스는 각각 한번씩 밖에 사용되고 있지 않아서 아쉬운 것 같네요. 개인적으로는 인터페이스를 줄이는 방향이 지금은 좋지 않을까 싶습니다.

Reply:
좋은 의견 감사합니다. `Piece`를 인자로 받는 케이스가 한번씩만 사용되어서 아쉽다는 의견에는 충분히 동의하지만, 인터페이스를 줄여서 얻는 이득이 무엇인지 잘 와닿지 않아서 지금 당장 `Piece`를 인자로 받는 메시지를 삭제해야 할 동기를 모르겠습니다.

```suggestion
    public boolean isSameTeam(Team moveTeam) {
        return team.equals(moveTeam);
    }

    public boolean isSamePosition(Position position) {
        return this.position.equals(position);
    }
```

예를 들어서 위와 같이 변경한다면, `GameBoard` 에서 두 개의 `Piece`가 같은 팀인지 확인하기 위해 
`piece.isSameTeam(killerPiece.getTeam());` 을 해야 합니다.
`Piece` 의 상태인 `Team`을 굳이 외부로 꺼내와서 다시 `Piece`에게 넘기는 것보다는, `Piece` 자체를 보내서 해당 로직 처리의 책임을 `Piece`에게 옮기고 `GameBoard`의 책임의 구체적인 부분을 조금 덜어내는 것도 좋지 않을까요?

```java
    private void killPieceBy(Piece killerPiece) {
        Optional<Piece> willKilledPiece = pieces.stream()
                .filter(piece -> piece.isSamePosition(killerPiece) && !piece.isSameTeam(killerPiece))
                .findFirst();

        willKilledPiece.ifPresent(pieces::remove);
    }
```

Reply:
클라이언트 코드 입장에서는 그렇게 볼 수도 있습니다. 하지만 Piece 입장에서 코드를 본다면 이미 있는 기능을 중복으로 만든 것 같다는 생각이 들어서 피드백을 남겼습니다. 딱히 현재 상황에서 어느 방향을 택해도 문제될 건 없다고 생각이 들긴하는데, 개인적 취향을 말해봤습니다. 히스타의 생각도 충분히 타당해서 원하는 방향으로 진행하시죠!